templates/support/SecurityClient.swift.hbs

```diff
@@ -0,0 +1,361 @@
+    ///   - password: A string representing the password for authentication.
+    public init(username: String, password: String) {
+        self.username = username
+        self.password = password
+    }
+    
+    public func authenticationFailed(securityScheme: SecurityScheme, scopes: [String]?) async throws {
+        
+    }
+    
+    public func authorize(securityScheme: SecurityScheme, scopes: [String]?, request: URLRequest) async throws -> URLRequest {
+        var request = request
+        let token = "\(username):\(password)".data(using: .utf8)!.base64EncodedString()
+        request.setValue("Basic \(token)", forHTTPHeaderField: "Authorization")
+        return request
+    }
+
+}
+
+/// A `SecurityClient` responsible for handling api key authentication credentials.
+public class APIKeySecurityClient: SecurityClient {
+    
+    private let apiKey: String
+    private let paramName: String
+    
+    public enum Mode {
+        case header
+        case query
+        case cookie
+    }
+    
+    private let mode: Mode
+    
+    /// Initializes a `APIKeySecurityClient` instance with provided credentials.
+    ///
+    /// - Parameters:
+    ///   - apiKey: A string representing the username for authentication.
+    ///   - mode: A setting to configure the location of the API Key in the request.
+    ///   - paramName: The name of the parameter which will appear as the header or query name
+    public init(apiKey: String, mode: Mode, paramName: String) {
+        self.apiKey = apiKey
+        self.mode = mode
+        self.paramName = paramName
+    }
+    
+    public func authenticationFailed(securityScheme: SecurityScheme, scopes: [String]?) async throws {
+        
+    }
+    
+    public func authorize(securityScheme: SecurityScheme, scopes: [String]?, request: URLRequest) async throws -> URLRequest {
+        var request = request
+        switch mode {
+        case .header:
+            request.setValue(apiKey, forHTTPHeaderField: paramName)
+        case .query:
+            guard let url = request.url else {
+                throw APIError.authenticationPrevented
+            }
+            var components = URLComponents(string: url.absoluteString)
+            var queryItems = components?.queryItems ?? []
+            queryItems.append(URLQueryItem(name: paramName, value: apiKey))
+            components?.queryItems = queryItems
+            request.url = components?.url
+        case .cookie:
+            let cookieProperties: [HTTPCookiePropertyKey: Any] = [
+                HTTPCookiePropertyKey.name: paramName,
+                HTTPCookiePropertyKey.value: apiKey,
+            ]
+
+            if let cookie = HTTPCookie(properties: cookieProperties) {
+                let headers = HTTPCookie.requestHeaderFields(with: [cookie])
+                headers.forEach { key, value in
+                    request.setValue(value, forHTTPHeaderField: key)
```

Comment:
I think we need to use `addValue` rather than `setValue` here. There might be multiple cookies and they all use the header name `"Cookie"`.